habitat-lab/habitat/datasets/rearrange/samplers/object_sampler.py

```diff
@@ -271,11 +272,8 @@ def sample_placement(
                         habitat_sim.utils.common.random_quaternion()
                     )
             if isinstance(receptacle, TriangleMeshReceptacle):
-                new_object_height = (
-                    new_object.root_scene_node.cumulative_bb.size()[1]
-                )
                 new_object.translation = new_object.translation + mn.Vector3(
-                    0, new_object_height / 2, 0
+                    0, 0.05, 0
```

Comment:
Why can you assume a fixed offset? Don't you need to adjust by the object height?

Reply:
Alex and I had a discussion about this, and the issue seems to be with the receptacle surface annotations clipping a bit into the receptacle mesh, and it's unrelated to the object height. Quoting Alex's response here:

> I agree that in this case moving the object first is not likely to cause an issue, but I don't think this is generally true. For example, objects in a drawer or on a shelf could be bumped up inside of something else.
The [snap_down](https://github.com/facebookresearch/habitat-lab/blob/864d6b503b61806fb7370c68453448dece593704/habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py#L231) function should already account for the height of the object. It projects the COM downward and then adjusts for the object's size. The only case in which this would be a problem is if the mesh receptacle surface is actually inside of the collision shape. Maybe we should check for this issue instead.
Alternatively, I might be more comfortable with a small fixed scalar offset for the COM, something like 10cm, which is very unlikely to push an object's COM into another part of the mesh.